File: RB101_programming_foundations/lesson_5/5_practice_problems/ex_14.rb
# Given this data structure write some code to return an array containing the colors of the fruits and the sizes of the vegetables. The sizes should be uppercase and the colors should be capitalized.

# The return value should look like this:

# [["Red", "Green"], "MEDIUM", ["Red", "Green"], ["Orange"], "LARGE"]

hsh = {
  'grape' => {type: 'fruit', colors: ['red', 'green'], size: 'small'},
  'carrot' => {type: 'vegetable', colors: ['orange'], size: 'medium'},
  'apple' => {type: 'fruit', colors: ['red', 'green'], size: 'medium'},
  'apricot' => {type: 'fruit', colors: ['orange'], size: 'medium'},
  'marrow' => {type: 'vegetable', colors: ['green'], size: 'large'},
}

hsh.each_with_object([]) do |(_, value), array|
  if value[:type] == 'fruit'
     array << value[:colors].map(&:capitalize)
  elsif value[:type] == 'vegetable'
    array << value[:size].upcase
  end
end

File: JS101_programming_foundations/lesson_5/practice_problems/ex_14.js
// Given the following data structure write some code to return an array containing the colors of the fruits and the sizes of the vegetables.The sizes should be uppercase, and the colors should be capitalized.

let obj = {
  grape: { type: 'fruit', colors: ['red', 'green'], size: 'small' },
  carrot: { type: 'vegetable', colors: ['orange'], size: 'medium' },
  apple: { type: 'fruit', colors: ['red', 'green'], size: 'medium' },
  apricot: { type: 'fruit', colors: ['orange'], size: 'medium' },
  marrow: { type: 'vegetable', colors: ['green'], size: 'large' },
};

let result = Object.values(obj).map(value => {
  if (value['type'] === 'fruit') {
    return value['colors'].map(word => word[0].toUpperCase() + word.slice(1));
  } else if (value['type'] === 'vegetable') {
    return value['size'].toUpperCase();
  }
});

console.log(result);